auth_token/models.py

```diff
@@ -129,55 +171,357 @@ def delete(self):
-        is_created_new_device_key = self.get_or_create(
-            uuid=uuid, user=user,
-            defaults={
-                'login_token': make_password(token),
-                'user_agent': user_agent[:256],
-                'is_active': True
-            }
-        )[1]
-        if is_created_new_device_key:
-            return token
+        secret_password = generate_key(length=64)
+        defaults = {
+            'login_token': make_password(secret_password),
+            'user_agent': user_agent[:256],
+            'is_active': True,
+            'name': name,
+            'slug': slug,
+            'is_primary': is_primary
+        }
+        mobile_device, is_created_mobile_device = self.get_or_create(
+            uuid=uuid,
+            user=user,
+            defaults=defaults
+        )
+        mobile_device.secret_password = secret_password
+        if not is_created_mobile_device and not mobile_device.is_active:
+            return mobile_device.change_and_save(**defaults)
+        elif is_created_mobile_device:
+            return mobile_device
         else:
-            raise DeviceKeyAlreadyExistsException('Device key already exists')
+            raise MobileDeviceAlreadyExists('Device key already exists')
 
 
-class DeviceKey(models.Model):
+class MobileDevice(SmartModel):
     """Model used to authenticate mobile devices. Unhashed login_token is stored
     in the device keychain and serve as password to log in together with UUID via DeviceBackend."""
 
-    created_at = models.DateTimeField(auto_now_add=True, null=False, blank=False, verbose_name=_('created at'))
     # this is not UUIDField because of the strict length limitation
-    uuid = models.CharField(verbose_name=_('UUID'), max_length=32)
-    last_login = models.DateTimeField(null=True, blank=True, verbose_name=_('last login'))
-    user = models.ForeignKey(django_settings.AUTH_USER_MODEL, on_delete=models.CASCADE, verbose_name=_('user'))
-    login_token = models.CharField(max_length=128, verbose_name=_('login token'))
-    is_active = models.BooleanField(default=True, verbose_name=_('is active'))
-    user_agent = models.CharField(max_length=256, null=True, blank=True, verbose_name=_('user agent'))
+    uuid = models.CharField(
+        verbose_name=_('UUID'),
+        max_length=36,
+        null=False,
+        blank=False
+    )
+    name = models.CharField(
+        verbose_name=_('name'),
+        max_length=250,
+        null=True,
+        blank=True
+    )
+    slug = models.SlugField(
+        verbose_name=_('slug'),
+        null=True,
+        blank=True
+    )
+    last_login = models.DateTimeField(
+        verbose_name=_('last login'),
+        null=True,
+        blank=True
+    )
+    user = models.ForeignKey(
+        to=django_settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        verbose_name=_('user'),
+        related_name='mobile_devices'
+    )
+    login_token = models.CharField(
```

Comment:
Maybe `hashed_password`?

Reply:
this is not password this is token, because we generate it. And I didn't want to name this valuse with hash prefix. Django doesn't do it too